ethkey/src/secret.rs

```diff
@@ -234,6 +246,35 @@ impl Deref for Secret {
 	}
 }
 
+// Wrapper to implement `AsMut` for `H256`.
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+struct H256Mut { val: H256 }
+
+impl H256Mut {
+	fn new(val: H256) -> H256Mut {
+		H256Mut { val }
+	}
+}
+
+impl AsRef<[u8]> for H256Mut {
+	fn as_ref(&self) -> &[u8] {
+		&self.val.0
+	}
+}
+
+impl AsMut<[u8]> for H256Mut {
```

Comment:
Is lack of this `impl`  the only reason you created `H256Mut`? I see no reason why it shouldn't be added to [paritytech/primitives](https://github.com/paritytech/primitives). Please make a pr with this change to that repo :)

Reply:
Yes, will do.

Reply:
TBH I don't see if it fits primitives. All primitives we have are immutable, do we want to have both mutable and immutable ones there? IMHO makes the library unnecessary bloated.

Reply:
There is already `DerefMut` in primitives though.

Reply:
Ah, ok. I thought we want to introduce `H256Mut` struct into that repo. If it's only about adding missing `AsMut` trait implementation then it's all good!